### PR TITLE
fix obsview for cuarrays

### LINKED ISSUE #72: `CuArray`s should not follow fast paths in `obsview`
`CuArray`s should not follow the fast path in `obsview` that returns a view of the original array, since this triggers scalar indexing for downstream operations.

ref https://github.com/FluxML/Flux.jl/issues/1935

### PR DESCRIPTION
**Pull request by CarloLucibello**
now `obsview` wraps generic abstract arrays into `ObsView` and returns a view instead only for Arrays.
~This fixes  #72~, and fixes  https://github.com/FluxML/Flux.jl/issues/1935
```julia
julia> using MLUtils, CUDA

julia> obsview([1,2,3]) # create a view with Array
3-element view(::Vector{Int64}, 1:3) with eltype Int64:
 1
 2
 3

julia> yg = rand(5) |> cu
5-element CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}:
 0.48661163
 0.8954821
 0.33829638
 0.1257987
 0.04116317

julia> obsview(yg) # wraps into ObsView with CuArrays
ObsView(::CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}, ::UnitRange{Int64})
 5 observations

julia> obsview(yg)[1:2]
ObsView(::CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}, ::UnitRange{Int64})
 2 observations

julia> getobs(obsview(yg), 1:2)
2-element CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}:
 0.60298043
 0.37703645


julia> getobs(shuffleobs(yg), 1:2)
2-element CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}:
 0.37703645
 0.58555883
```

Edit:
In the end, I opted for a less invading change only in the DataLoader

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 name = "MLUtils"
 uuid = "f1d291b0-491e-4a28-83b9-f70985020b54"
 authors = ["Carlo Lucibello <carlo.lucibello@gmail.com> and contributors"]
-version = "0.2.2"
+version = "0.2.3"
 
 [deps]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
```
Comment (ToucheSir) [inline Project.toml:7]: Looks this this dep isn't required after all?

File: src/obsview.jl [hunk 2]
```diff
@@ -221,7 +221,10 @@ obsview(data, indices=1:numobs(data)) = ObsView(data, indices)
 
 obsview(A::SubArray) = A
 
-function obsview(A::AbstractArray{T,N}, idx) where {T,N}
+## We dont't relax this to AbstractArray due to 
+# https://github.com/FluxML/Flux.jl/issues/1935
+# https://github.com/JuliaML/MLUtils.jl/issues/72
+function obsview(A::Array{T,N}, idx) where {T,N}
```
Comment (oschulz) [inline src/obsview.jl:227]: Maybe allow CPU subarrays as well?

```julia
function obsview(A::Union{Array{T,N},SubArray{T,N,<:Array}}, idx) where {T,N}
...
```

Comment (CarloLucibello) [reply src/obsview.jl:227]: SubArrays are already taken care of
```julia
julia> obsview([1,2,3])
3-element view(::Vector{Int64}, 1:3) with eltype Int64:
 1
 2
 3

julia> obsview(obsview([1,2,3]))
3-element view(::Vector{Int64}, 1:3) with eltype Int64:
 1
 2
 3
```

Comment (CarloLucibello) [reply src/obsview.jl:227]: sorry, that's not true
```julia
julia> obsview(obsview([1,2,3], 1:2), 1:2)
ObsView(view(::Vector{Int64}, 1:2), ::UnitRange{Int64})
 2 observations
```

Comment (CarloLucibello) [reply src/obsview.jl]: done

**Comment by CarloLucibello** [comment]
Trying to special case cpu arrays only for returning a `view`  instead of a `ObsView` turned out to be a rabbit hole, you have Array, SparseMatrixCSC, Adjoint, Transpose,... and also create an inconsistent experience for the users.

So in the end I just made the minimally invasive change in `DataLoader` just fixing https://github.com/FluxML/Flux.jl/issues/1935

**Comment by oschulz** [comment]
Thanks for the quick fix @CarloLucibello !

**Pull request merged**